tests/testthat/test-merge_sce_list.R

```diff
@@ -298,9 +303,124 @@ test_that("merging SCEs without names works as expected", {
+      sce_list
+      )
+  )
+})
+
+
+# Tests with altexps -------------------------------------------------------
+
+
+## helper function to add an altExp to a simulated SCE ----
+add_sce_altexp <- function(
+    sce,
+    batch,
+    num_altexp_features,
+    n_cells) {
+
+  sce_alt <- sim_sce(
+    n_genes = num_altexp_features,
+    n_cells = n_cells,
+    n_empty = 0)
+
+  # ensure matching barcodes
+  colnames(sce_alt) <- colnames(sce)
+
+  # add some rowdata columns
+  rowData(sce_alt)[["feature_column"]] <- rownames(sce_alt)
+  rowData(sce_alt)[["other_column"]] <- runif(nrow(sce_alt))
+
+  # add a coldata columns
+  colData(sce_alt)[["coldata_column"]] <- runif(ncol(sce_alt))
+
+  # add logcounts
+  logcounts(sce_alt) <- counts(sce_alt)
+
+  # add metadata
+  library_id <- glue::glue("library-{batch}")
+  sample_id <- glue::glue("sample-{batch}")
+
+  metadata(sce_alt)$library_id <- library_id
+  metadata(sce_alt)$sample_id <- sample_id
+  metadata(sce_alt)$mapped_reads <- 100
+
+  # add sce_alt as sce's altExp
+  altExp(sce) <- sce_alt
+
+  return(sce)
+}
+
+sce_list_with_altexp <- sce_list |>
+  purrr::imap(
+    add_sce_altexp,
+    num_altexp_features,
+    total_cells / 3
+  )
+# keep only the first 3 features from the first SCE
+altExp(sce_list_with_altexp[[1]]) <- altExp(sce_list_with_altexp[[1]])[1:3]
+
+# vector of all expected names
+full_altexp_names <- names(altExp(sce_list_with_altexp[[2]]))
+
+# set altExp names themselves - all altexp are the same name here
+altexp_name <- "adt"
+sce_list_with_altexp <- sce_list_with_altexp |>
+  purrr::map(
+    \(sce) {
+      altExpNames(sce) <- altexp_name
+      sce # return the sce
+    }
+  )
+
+test_that("merging SCEs with matching features works as expected, with altexps", {
+
+  merged_sce <- merge_sce_list(
+    sce_list_with_altexp,
+    batch_column = batch_column,
+    # "total" should get removed
+    retain_coldata_cols = retain_coldata_cols,
+    # this row name should not be modified:
+    preserve_rowdata_cols = c("gene_names")
+  )
+
+  merged_altexp <- altExp(merged_sce)
+
+
+  expect_true(altExpNames(merged_sce) == altexp_name)
+  expect_equal(dim(merged_altexp), c(num_altexp_features, total_cells))
+  expect_equal(rownames(merged_altexp), full_altexp_names)
+
+  expected_colnames <- sce_list_with_altexp |>
+    purrr::imap(
+      \(sce, sce_name) glue::glue("{sce_name}-{colnames(sce)}")
+    ) |>
+    unlist() |>
+    unname()
+  expect_equal(colnames(merged_altexp), expected_colnames)
+
+})
+
+
+test_that("merging SCEs with altExps works as expected when include_altexps = FALSE", {
+
+  merged_sce <- merge_sce_list(
+    sce_list_with_altexp,
+    batch_column = batch_column,
+    # "total" should get removed
+    retain_coldata_cols = retain_coldata_cols,
+    # this row name should not be modified:
+    preserve_rowdata_cols = c("gene_names"),
+    include_altexp = FALSE
+  )
+
+  # this should fail if no altExp exists
+  expect_error( altExp(merged_sce) )
```

Comment:
A slightly more specific test (other errors won't silently pass)
```suggestion
  # there should not be any altExps
  expect_length(altExpNames(merged_sce), 0)
```
```